File: src/pages/mailto.py
'''
Created on 30 Jul 2013

@author: hicksj
'''

class Mailto:
    '''
    classdocs
    '''


    def __init__(self, userid, linkText, domain="", description=""):
        '''
        Constructor
        '''
        html = """<script language="javascript">
            document.write(mailTo("{theId}", "{theDomain}", "{theDescription}", "{theLinkText}"));
        </script>
        <noscript>{theLinkText}<i> (Javascript not enabled: cannot display mail link)</i>
        </noscript>"""
        self.html = html.format(theId = userid, theDomain = domain, theDescription = description, theLinkText = linkText)

File: website/html/mailto.php
<?php
class Mailto
{
	private $id;
	private $domain;
	private $description = null;
	
	function __construct($id, $domain=null)
	{
		$this->id = $id;
		$this->domain = $domain;
	}
	
	function setDescription($description)
	{
		$this->description = $description;
		return $this;
	}
	
	function getMailtoCall($linkText=null)
	{
		$dom = $this->domain == null ? "\"\"" : "\"$this->domain\"";
		$descr = $this->description == null ? "\"\"" : "\"$this->description\"";
		$link = $linkText == null ? "\"\"" : "\"$linkText\"";
		return "<script>document.write(mailTo(\"$this->id\", $dom, $descr, $link));</script>";
	}
}